From 119157a42388b9db569341700555785767cc0a5d Mon Sep 17 00:00:00 2001
From: Neena Busireddy <neenareddy.busireddy@nxp.com>
Date: Thu, 14 Jun 2018 10:59:45 -0500
Subject: [PATCH] Update EglFSScreen to Qt 5.9

In Qt 5.9, we ported the mode API to QPlatformScreen so the original
code should not be built with that version.

Signed-off-by: Neena Busireddy <neenareddy.busireddy@nxp.com>
---
 plugins/egldeviceintegration/kms/eglfskmsscreen.cpp | 14 ++------------
 plugins/egldeviceintegration/kms/eglfskmsscreen.h   |  3 +--
 src/platform/deviceintegration/eglfsscreen.cpp      | 10 ++++++----
 src/platform/deviceintegration/eglfsscreen.h        | 15 +++++++++------
 src/server/screen/nativescreenbackend.cpp           |  2 +-
 5 files changed, 19 insertions(+), 25 deletions(-)

diff --git a/plugins/egldeviceintegration/kms/eglfskmsscreen.cpp b/plugins/egldeviceintegration/kms/eglfskmsscreen.cpp
index bae783b..c50b1cf 100644
--- a/plugins/egldeviceintegration/kms/eglfskmsscreen.cpp
+++ b/plugins/egldeviceintegration/kms/eglfskmsscreen.cpp
@@ -466,9 +466,9 @@ void EglFSKmsScreen::setPowerState(EglFSScreen::PowerState state)
     m_powerState = state;
 }
 
-QList<EglFSScreen::Mode> EglFSKmsScreen::modes() const
+QVector<EglFSScreen::Mode> EglFSKmsScreen::modes() const
 {
-    QList<EglFSScreen::Mode> list;
+    QVector<EglFSScreen::Mode> list;
 
     Q_FOREACH (const drmModeModeInfo &info, m_output.modes)
         list.append({QSize(info.hdisplay, info.vdisplay),
@@ -502,16 +502,6 @@ int EglFSKmsScreen::preferredMode() const
     return m_output.preferred_mode;
 }
 
-void EglFSKmsScreen::setPreferredMode(int modeId)
-{
-    if (modeId < 0 || modeId >= m_output.modes.size()) {
-        qCWarning(lcKms, "Invalid mode passed to EglFSKmsScreen::setPreferredMode()");
-        return;
-    }
-
-    m_output.preferred_mode = modeId;
-}
-
 QString EglFSKmsScreen::identifier() const
 {
     return m_edid.identifier;
diff --git a/plugins/egldeviceintegration/kms/eglfskmsscreen.h b/plugins/egldeviceintegration/kms/eglfskmsscreen.h
index 52467f9..1cd34b6 100644
--- a/plugins/egldeviceintegration/kms/eglfskmsscreen.h
+++ b/plugins/egldeviceintegration/kms/eglfskmsscreen.h
@@ -128,13 +128,12 @@ public:
     EglFSScreen::PowerState powerState() const Q_DECL_OVERRIDE;
     void setPowerState(EglFSScreen::PowerState state) Q_DECL_OVERRIDE;
 
-    QList<EglFSScreen::Mode> modes() const Q_DECL_OVERRIDE;
+    QVector<EglFSScreen::Mode> modes() const Q_DECL_OVERRIDE;
 
     int currentMode() const Q_DECL_OVERRIDE;
     void setCurrentMode(int modeId) Q_DECL_OVERRIDE;
 
     int preferredMode() const Q_DECL_OVERRIDE;
-    void setPreferredMode(int modeId) Q_DECL_OVERRIDE;
 
     QString identifier() const Q_DECL_OVERRIDE;
     QString manufacturer() const Q_DECL_OVERRIDE;
diff --git a/src/platform/deviceintegration/eglfsscreen.cpp b/src/platform/deviceintegration/eglfsscreen.cpp
index 9a6ad39..910d385 100644
--- a/src/platform/deviceintegration/eglfsscreen.cpp
+++ b/src/platform/deviceintegration/eglfsscreen.cpp
@@ -228,6 +228,7 @@ QPixmap EglFSScreen::grabWindow(WId wid, int x, int y, int width, int height) co
     return QPixmap();
 }
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 9, 0)
 /*!
   Returns the list of modes.
 
@@ -237,9 +238,9 @@ QPixmap EglFSScreen::grabWindow(WId wid, int x, int y, int width, int height) co
   \sa QPlatformScreen::geometry
   \sa QPlatformScreen::refreshRate
 */
-QList<EglFSScreen::Mode> EglFSScreen::modes() const
+QVector<EglFSScreen::Mode> EglFSScreen::modes() const
 {
-    QList<EglFSScreen::Mode> list;
+    QVector<EglFSScreen::Mode> list;
     list.append({geometry().size(), refreshRate()});
     return list;
 }
@@ -255,7 +256,7 @@ int EglFSScreen::currentMode() const
 {
     return 0;
 }
-
+#endif
 /*!
   Sets the index of the current mode from the modes list.
 
@@ -268,6 +269,7 @@ void EglFSScreen::setCurrentMode(int modeId)
     Q_UNUSED(modeId);
 }
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 9, 0)
 /*!
   Returns the index of the preferred mode from the modes list.
 
@@ -291,7 +293,7 @@ void EglFSScreen::setPreferredMode(int modeId)
 {
     Q_UNUSED(modeId);
 }
-
+#endif
 /*!
   Returns the screen identifier.
 
diff --git a/src/platform/deviceintegration/eglfsscreen.h b/src/platform/deviceintegration/eglfsscreen.h
index ed9cf10..c10b44b 100644
--- a/src/platform/deviceintegration/eglfsscreen.h
+++ b/src/platform/deviceintegration/eglfsscreen.h
@@ -51,11 +51,12 @@ class EglFSWindow;
 class LIRIPLATFORM_EXPORT EglFSScreen : public QPlatformScreen
 {
 public:
+#if QT_VERSION < QT_VERSION_CHECK(5, 9, 0)
     struct Mode {
         QSize size;
         qreal refreshRate;
     };
-
+#endif
     EglFSScreen(EGLDisplay display);
     ~EglFSScreen();
 
@@ -83,15 +84,17 @@ public:
     void setPrimarySurface(EGLSurface surface);
 
     void handleCursorMove(const QPoint &pos);
-
-    virtual QList<Mode> modes() const;
-
+#if QT_VERSION < QT_VERSION_CHECK(5, 9, 0)
+    virtual QVector<Mode> modes() const;
+#endif
+#if QT_VERSION < QT_VERSION_CHECK(5, 9, 0)
     virtual int currentMode() const;
+#endif
     virtual void setCurrentMode(int modeId);
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 9, 0)
     virtual int preferredMode() const;
-    virtual void setPreferredMode(int modeId);
-
+#endif
     virtual QString identifier() const;
     virtual QString manufacturer() const;
     virtual QString model() const;
diff --git a/src/server/screen/nativescreenbackend.cpp b/src/server/screen/nativescreenbackend.cpp
index e7e6fb9..302fa27 100644
--- a/src/server/screen/nativescreenbackend.cpp
+++ b/src/server/screen/nativescreenbackend.cpp
@@ -157,7 +157,7 @@ void NativeScreenBackend::handleScreenChanged(QScreen *qscreen, Screen *screen)
             static_cast<Platform::EglFSScreen *>(qscreen->handle());
     if (QGuiApplication::platformName() == QLatin1String("liri") && eglfsScreen) {
         QList<Screen::Mode> modes;
-        Q_FOREACH (const Platform::EglFSScreen::Mode &mode, eglfsScreen->modes())
+        for (const auto &mode : eglfsScreen->modes())
             modes.append({mode.size, mode.refreshRate});
         screenPrivate->setModes(modes);
 
-- 
1.9.1

